src/generateCanonicalFieldsInserts.ts

```diff
@@ -28,16 +28,27 @@ psvInput.pipe(
     delimiter: '|',
     asObject: true,
   }),
-).on('data', (row: any) => {
-  const label = row.label.replace('\'', '\'\'');
-  const shortCode = row.shortCode.replace('\'', '\'\'');
-  const dataType = row.dataType.replace('\'', '\'\'');
-  const createdAt = row.createdAt.replace('\'', '\'\'');
+).on('data', (row: Object) => {
+  const record = Object.values(row)[0];
+  //regex to remove extra quotes and correct quotes hierarchy i.e. \" inside datatype string enclosed by \' 
+  const string = record.replace(/\"/g,'\'').replace(/\'\'/g,'\"')
+  //split on the basis of comma but don't consider commas inside quotes
+  const values = string.match(/('.*?'|[^',\s]+)(?=\s*,|\s*$)/g)
+
+  const id = values[0];
+  let label = values[1];
+  const shortCode = values[2];
+  const dataType = values[3];
+
+  //remove enclosing single qoutes of label field having comma separated value 
+  if( label[0] === '\''){
+    label = label.slice(1,-1)
+  }
   if (firstRowArrived) {
-    sqlOutput.write(`,${os.EOL}(${row.id}, '${label}', '${shortCode}', '${dataType}', '${createdAt}')`);
+    sqlOutput.write(`,${os.EOL}(${id}, '${label}' , '${shortCode}', ${dataType} )`);
```

Comment:
Instead of setting variables `id`, `label`, etc, I think these can become `row.ID`, `row.Label`, `row['Internal field name']`, etc. But I think I see the quotes issue with `row.Type`. Perhaps the variables are helpful after all but they should be passed through a helper function rather than sending the whole row through the regex?